runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java

```diff
@@ -75,10 +75,15 @@
 	private static final Pattern BINDING_USED_PATTERN = Pattern.compile(
 		"You bind the temple's power into (mud|lava|steam|dust|smoke|mist) runes\\.");
 	private static final String BINDING_BREAK_TEXT = "Your Binding necklace has disintegrated.";
+	private static final Pattern RING_OF_FORGING_CHECK_PATTERN = Pattern.compile(
+		"You can smelt ([0-9+]+|one) more pieces of iron ore before a ring melts\\.");
```

Comment:
when only one charge remains pieces -> piece 

![](https://cdn.discordapp.com/attachments/419891709883973642/601102106396459036/unknown.png)

Reply:
Thank you! I added the greedy quantifier '?' we see in the other pattern matchers when the syntax changes on the last charge.

Reply:
You should be adding test cases which contain the actual text the game uses. At least one of your current test cases is just made up.